src/write_to_file.cc

```diff
@@ -47,20 +51,22 @@ int writeValueToFile(
 }
 
 
-void writeVectorToFile(std::vector<double> v, std::string pFilename){
+void writeVectorToFile(std::vector<double> v, fs::path filename){
```

Comment:
THIS one actually needs to be modifyable in the current implementation (see lines 59-61). We add some stuff to the filename internally.